examples/integrations/oci/atp-reactive/src/main/java/io/helidon/examples/integrations/oci/atp/reactive/AtpService.java

```diff
@@ -69,30 +114,85 @@ private void generateWallet(ServerRequest req, ServerResponse res) {
+                    .orElseThrow(() -> new IllegalStateException("Missing tnsNetServiceName!!"))));
             pds.setUser(config.get("db.userName").as(String.class).orElse("ADMIN"));
             pds.setPassword(config.get("db.password")
-                    .as(String.class)
-                    .orElseThrow(() -> new IllegalStateException("Missing password!!")));
+                                    .as(String.class)
+                                    .orElseThrow(() -> new IllegalStateException("Missing password!!")));
             pds.setConnectionFactoryClassName(OracleDataSource.class.getName());
         } catch (SQLException e) {
             LOGGER.log(Level.SEVERE, "Error setting up PoolDataSource", e);
             return Single.error(e);
         }
         return Single.just(new JdbcDbClientProvider().builder()
-                .connectionPool(() -> {
-                    try {
-                        return pds.getConnection();
-                    } catch (SQLException e) {
-                        throw new IllegalStateException("Error while setting up new connection", e);
-                    }
-                })
-                .build());
+                                   .connectionPool(() -> {
+                                       try {
+                                           return pds.getConnection();
+                                       } catch (SQLException e) {
+                                           throw new IllegalStateException("Error while setting up new connection", e);
+                                       }
+                                   })
+                                   .build());
+    }
+
+    /**
+     * Returns SSLContext based on cwallet.sso in wallet.
+     *
+     * @return SSLContext
+     */
+    private static SSLContext getSSLContext(byte[] walletContent) throws IllegalStateException {
+        SSLContext sslContext = null;
+        try (ZipInputStream zis = new ZipInputStream(new BufferedInputStream(new ByteArrayInputStream(walletContent)))) {
+            ZipEntry entry = null;
+            while ((entry = zis.getNextEntry()) != null) {
+                if (entry.getName().equals("cwallet.sso")) {
+                    KeyStore keyStore = KeyStore.getInstance("SSO", new OraclePKIProvider());
+                    keyStore.load(zis, null);
+                    TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance("PKIX");
+                    KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("PKIX");
+                    trustManagerFactory.init(keyStore);
+                    keyManagerFactory.init(keyStore, null);
+                    sslContext = SSLContext.getInstance("TLS");
+                    sslContext.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), null);
+                }
+                zis.closeEntry();
```

Comment:
Maybe needs to be in a `finally` block?  I haven't used the zip-oriented stuff in the JDK in a long time.

Reply:
This is to close entry that was created in try block. ZIS is created in try-with-resources so should be handled by that.